File: 2019W45/preppindataw45.R
library(dplyr)
library(readr)
library(purrr)
library(stringr)

rowdata <- dir("E:/InputW45/", pattern = "\\w+\\s\\w+\\.csv") %>%
  map(~ read_csv(file.path("E:/InputW45/", .),col_types =cols(`Scent` = col_character(),
                                                              `Sales Value` = col_double(),
                                                              `Sales Volume` = col_double())) 
      %>% mutate('Dummy'=.x,'Store' = str_extract(.x,'^(\\w+)'),
                 'Weekday' = str_extract(.x,'(\\w+)(?=\\.)'))) %>%
  reduce(rbind) %>%
  distinct() %>%
  merge(., read_csv("E:/InputW45/Dates.csv",col_types =cols(`Dates` = col_character())) %>%
          mutate('Weekday' = str_extract(`Dates`,'^(\\w+)')), on='Weekday')
  
final1 <- rowdata %>%
  group_by(`Store`,`Scent`) %>%
  summarise('Sales Volume' = sum(`Sales Volume`),
            'Sales Value' = sum(`Sales Value`)) %>%
  group_by(`Store`) %>%
  mutate('Scent % of Store Sales Volumes' = round(`Sales Volume`/sum(`Sales Volume`),2),
         'Scent % of Store Sales Values' = round(`Sales Value`/sum(`Sales Value`),2)) %>%
  select(-c('Sales Volume','Sales Value'))

final2 <- rowdata %>%
  group_by(`Store`,`Dates`) %>%
  summarise('Sales Volume' = sum(`Sales Volume`),
            'Sales Value' = sum(`Sales Value`)) %>%
  group_by(`Store`) %>%
  mutate('Weekday % of Store Sales Volumes' = round(`Sales Volume`/sum(`Sales Volume`),2),
         'Weekday % of Store Sales Values' = round(`Sales Value`/sum(`Sales Value`),2)) %>%
  select(-c('Sales Volume','Sales Value'))

View(final1)
View(final2)

File: 2019W45/PreppindataW45.py
import pandas as pd
import os
import re

os.chdir(r'E:/InputW45/')
mergedata = []
for files in [f for f in os.listdir('.') if re.match('\w+\s\w+.csv', f)]:
    dataset = pd.read_csv(files)
    dataset['Store']= re.search('^\w+(?=\s)',files).group(0)
    dataset['Weekday']= re.search('\w+(?=\.)',files).group(0)
    mergedata.append(dataset)    
sales = pd.concat(mergedata).drop_duplicates()   

dates = pd.read_csv('Dates.csv')
dates['Weekday'] = dates['Dates'].apply(lambda x: re.search('^\w+(?=\s)',x).group(0))

rowdata = pd.merge(sales,dates,how='inner',on='Weekday')

final1 = rowdata.groupby(['Store','Scent']).agg({'Sales Value':'sum', 'Sales Volume':'sum'})
final1['Scent % of Store Sales Volumes'] = final1['Sales Volume'].groupby(level=0).apply(lambda x: round(x/float(x.sum()),2))
final1['Scent % of Store Sales Values'] = final1['Sales Value'].groupby(level=0).apply(lambda x: round(x/float(x.sum()),2))
final1 = final1.reset_index(drop=False).drop(['Sales Volume','Sales Value'],axis=1)

final2 = rowdata.groupby(['Store','Dates']).agg({'Sales Value':'sum', 'Sales Volume':'sum'})
final2['Weekday % of Store Sales Volumes'] = final2['Sales Volume'].groupby(level=0).apply(lambda x: round(x/float(x.sum()),2))
final2['Weekday % of Store Sales Values'] = final2['Sales Value'].groupby(level=0).apply(lambda x: round(x/float(x.sum()),2))
final2 = final2.reset_index(drop=False).drop(['Sales Volume','Sales Value'],axis=1)
